### PR TITLE
Integration tests for Organization /meta-attribute GET API

### PR DESCRIPTION
**Pull request by HasiniSama**
### Purpose
$Subject

**Main Tests**
- testGetOrganizationsMetaAttributesWithInvalidOperator()
- testGetOrganizationsMetaAttributesWithUnsupportedAttribute()
- testGetOrganizationsMetaAttributes(String filterQuery, boolean isRecursive, boolean expectEmptyList)

**Support Tests**
- testOnboardChildOrganization()
- testDeleteChildOrganization()

### Related Issues
- https://github.com/wso2/product-is/issues/20611

**Merge After**
> https://github.com/wso2/identity-organization-management-core/pull/144
> https://github.com/wso2/identity-api-server/pull/628

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementFailureTest.java [hunk 1]
```diff
@@ -114,6 +114,7 @@ public void testGetM2MAccessToken() throws Exception {
     public void testSelfOnboardOrganization(String requestBodyPath) throws Exception {
 
         String body = readResource(requestBodyPath);
+        body = body.replace("${parentId}", "");
```
Comment (AnuradhaSK) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementFailureTest.java:117]:
```suggestion
        body = body.replace("${parentId}", StringUtils.Empty);
```

Comment (HasiniSama) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementFailureTest.java]: This is updated in all places.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java [hunk 2]
```diff
@@ -381,6 +387,64 @@ public void unShareB2BApplication() throws JSONException {
...
+                {"?filter=attributes eq Country&limit=10&recursive=false", false, false},
+                {"?filter=attributes eq Region&limit=10&recursive=true", true, false},
+                {"?filter=attributes sw C and attributes ew try&limit=10&recursive=false", false, false},
+                {"?filter=attributes co A&limit=10&recursive=true", false, true},
+        };
+    }
+
+    @Test(dependsOnMethods  = "testOnboardChildOrganization",
+          dataProvider      = "dataProviderForGetOrganizationsMetaAttributes")
+    public void testGetOrganizationsMetaAttributes(String filterQuery, boolean isRecursive, boolean expectEmptyList) {
```
Comment (AnuradhaSK) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java:424]: other than the filter query, limit and recursive query params have been passed to the first
Can we fix the data provider?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java [hunk 3]
```diff
@@ -381,6 +387,64 @@ public void unShareB2BApplication() throws JSONException {
...
+
+    @DataProvider(name = "dataProviderForGetOrganizationsMetaAttributes")
+    public Object[][] dataProviderForGetOrganizationsMetaAttributes() {
+
+        return new Object[][] {
+                {"?filter=attributes eq Country&limit=10&recursive=false", false, false},
+                {"?filter=attributes eq Region&limit=10&recursive=true", true, false},
+                {"?filter=attributes sw C and attributes ew try&limit=10&recursive=false", false, false},
+                {"?filter=attributes co A&limit=10&recursive=true", false, true},
+        };
```
Comment (AnuradhaSK) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java:419]: why recursive param is in first argument ? as per the test method signature 2nd argument is for that

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java (hunk 2 above)
Comment (HasiniSama) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java]: This is now updated to only pass the filter as the first param.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java (hunk 3 above)
Comment (HasiniSama) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java]: This is now updated in the in the last data.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java [hunk 4]
```diff
@@ -381,6 +387,64 @@ public void unShareB2BApplication() throws JSONException {
...
+    }
+
+    @DataProvider(name = "dataProviderForGetOrganizationsMetaAttributes")
+    public Object[][] dataProviderForGetOrganizationsMetaAttributes() {
+
+        return new Object[][] {
+                {"?filter=attributes eq Country&limit=10&recursive=false", false, false},
+                {"?filter=attributes eq Region&limit=10&recursive=true", true, false},
+                {"?filter=attributes sw C and attributes ew try&limit=10&recursive=false", false, false},
+                {"?filter=attributes co A&limit=10&recursive=true", false, true},
```
[the remark this answers is not in the document]
Comment (HasiniSama) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java]: This is now updated in the in the last data.

**Review by AnuradhaSK — APPROVED** [review]

**Pull request merged**